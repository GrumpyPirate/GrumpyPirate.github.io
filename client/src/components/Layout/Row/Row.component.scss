@import 'core';

.row {
  align-items: stretch;
  display: flex;
  flex-flow: row wrap;
  margin-left: rem(map-get($gutter-widths, 'xs') / -2);
  margin-right: rem(map-get($gutter-widths, 'xs') / -2);

  @each $breakpoint in map-keys($gutter-widths) {
    @if index($breakpoints, $breakpoint map-get($breakpoints, $breakpoint)) > 1 {
      @include respond-up($breakpoint) {
        margin-left: rem(map-get($gutter-widths, $breakpoint) / -2);
        margin-right: rem(map-get($gutter-widths, $breakpoint) / -2);
      }
    }
  }

  // Modifiers
  &--align {
    &-top { align-items: flex-start; }
    &-middle { align-items: center; }
    &-bottom { align-items: flex-end; }

    @each $breakpoint in map-keys($breakpoints) {
      @include respond-up($breakpoint) {
        &-#{$breakpoint}-top { align-items: flex-start; }
        &-#{$breakpoint}-middle { align-items: center; }
        &-#{$breakpoint}-bottom { align-items: flex-end; }
      }
    }
  }

  &--justify {
    &-left { justify-content: flex-start; }
    &-center { justify-content: center; }
    &-right { justify-content: flex-end; }
    &-space-between { justify-content: space-between; }

    @each $breakpoint in map-keys($breakpoints) {
      @include respond-up($breakpoint) {
        &-#{$breakpoint}-left { justify-content: flex-start; }
        &-#{$breakpoint}-center { justify-content: center; }
        &-#{$breakpoint}-right { justify-content: flex-end; }
        &-#{$breakpoint}-space-between { justify-content: space-between; }
      }
    }
  }
}
