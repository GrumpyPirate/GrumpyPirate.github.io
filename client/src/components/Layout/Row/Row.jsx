import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import classes from './Row.scss';

const Row = ({ align, justify, children }) => (
  <div
    className={classnames(
      classes.row, {
        [classes[`row--align-${align}`]]: align,
        [classes[`row--justify-${justify}`]]: justify,
      },
    )}
  >
    {children}
  </div>
);

Row.propTypes = {
  align: PropTypes.oneOf(['top', 'middle', 'bottom']),
  justify: PropTypes.oneOf(['left', 'center', 'right', 'space-between', 'space-around']),
  children: PropTypes.node,
};

Row.defaultProps = {
  align: null,
  justify: null,
  children: null,
};

export default Row;
