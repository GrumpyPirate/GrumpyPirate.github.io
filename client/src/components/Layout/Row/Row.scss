@import 'core';

.row {
  display: flex;
  flex-flow: row wrap;
  align-items: stretch;
  margin-right: rem(map-get($gutter-widths, xs) / -2);
  margin-left: rem(map-get($gutter-widths, xs) / -2);

  @each $breakpoint in map-keys($gutter-widths) {
    @if index($breakpoints, $breakpoint map-get($breakpoints, $breakpoint)) > 1 {
      @include respond-up($breakpoint) {
        margin-right: rem(map-get($gutter-widths, $breakpoint) / -2);
        margin-left: rem(map-get($gutter-widths, $breakpoint) / -2);
      }
    }
  }

  // Modifiers
  &--align {
    &-top {
      align-items: flex-start;
    }

    &-middle {
      align-items: center;
    }

    &-bottom {
      align-items: flex-end;
    }
  }

  &--justify {
    &-left {
      justify-content: flex-start;
    }

    &-center {
      justify-content: center;
    }

    &-right {
      justify-content: flex-end;
    }

    &-space-between {
      justify-content: space-between;
    }

    &-space-around {
      justify-content: space-around;
    }
  }
}
