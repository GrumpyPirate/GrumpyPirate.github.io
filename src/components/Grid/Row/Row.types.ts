export type RowXAlign = 'left' | 'center' | 'right';
export type RowYAlign = 'top' | 'center' | 'bottom' | 'stretch';

export interface RowProps {
  xAlign?: RowXAlign;
  yAlign?: RowYAlign;
}
