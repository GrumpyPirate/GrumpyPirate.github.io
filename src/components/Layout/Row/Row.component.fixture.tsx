import React from 'react';

import Column from 'components/Layout/Column/Column.component';

export default {
  children: (
    <Column>
      <div>Test</div>
    </Column>
  ),
};
