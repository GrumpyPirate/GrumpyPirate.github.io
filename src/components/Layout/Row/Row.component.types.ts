import { ChildrenProps } from 'types/common';

export interface RowProps extends ChildrenProps {
  align?: 'top' | 'middle' | 'bottom';
  alignSM?: 'top' | 'middle' | 'bottom';
  alignMD?: 'top' | 'middle' | 'bottom';
  alignLG?: 'top' | 'middle' | 'bottom';
  alignXL?: 'top' | 'middle' | 'bottom';
  justify?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
  justifySM?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
  justifyMD?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
  justifyLG?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
  justifyXL?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
}
