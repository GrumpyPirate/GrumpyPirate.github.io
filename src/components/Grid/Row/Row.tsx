import styled, { css } from 'styled-components';

import { gutterWidths, mediaQueries, rem } from 'styles';

import { RowProps } from './Row.types';

const Row = styled.div<RowProps>`
  display: flex;
  flex-flow: row wrap;
  margin-right: ${rem(gutterWidths.xs / -2)};
  margin-left: ${rem(gutterWidths.xs / -2)};

  ${({ xAlign }) => {
    switch (xAlign) {
      case 'center':
        return css`
          justify-content: center;
        `;
      case 'right':
        return css`
          justify-content: flex-end;
        `;
      default:
        return '';
    }
  }}

  ${({ yAlign }) => {
    switch (yAlign) {
      case 'top':
        return css`
          align-items: flex-start;
        `;
      case 'center':
        return css`
          align-items: center;
        `;
      case 'bottom':
        return css`
          align-items: flex-end;
        `;
      case 'stretch':
        return css`
          align-items: stretch;
        `;
      default:
        return '';
    }
  }}

  @media ${mediaQueries.md} {
    margin-right: ${rem(gutterWidths.md / -2)};
    margin-left: ${rem(gutterWidths.md / -2)};
  }
`;

export default Row;
