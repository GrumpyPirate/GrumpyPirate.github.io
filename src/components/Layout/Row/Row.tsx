import classnames from 'classnames';
import React, { FunctionComponent } from 'react';

import classes from './Row.scss';
import { RowProps } from './Row.types';

const Row: FunctionComponent<RowProps> = ({
  align,
  alignSM,
  alignMD,
  alignLG,
  alignXL,
  justify,
  justifySM,
  justifyMD,
  justifyLG,
  justifyXL,
  children,
}) => (
  <div
    className={classnames(classes.row, {
      [classes[`row--align-${align}`]]: align,
      [classes[`row--align-sm-${alignSM}`]]: alignSM,
      [classes[`row--align-md-${alignMD}`]]: alignMD,
      [classes[`row--align-lg-${alignLG}`]]: alignLG,
      [classes[`row--align-xl-${alignXL}`]]: alignXL,
      [classes[`row--justify-${justify}`]]: justify,
      [classes[`row--justify-sm-${justifySM}`]]: justifySM,
      [classes[`row--justify-md-${justifyMD}`]]: justifyMD,
      [classes[`row--justify-lg-${justifyLG}`]]: justifyLG,
      [classes[`row--justify-xl-${justifyXL}`]]: justifyXL,
    })}
  >
    {children}
  </div>
);

export default Row;
