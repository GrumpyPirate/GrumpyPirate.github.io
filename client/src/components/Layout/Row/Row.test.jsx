import React from 'react';
import Row from './Row';

describe('<Row />', () => {
  const props = {
    children: <div>Test children</div>,
  };

  it('should exist', () => {
    expect(Row).toBeDefined();
  });

  it('should be a function', () => {
    expect(typeof Row).toBe('function');
  });

  it('should render as expected, without crashing', () => {
    expect(<Row {...props} />).toMatchSnapshot();
  });
});
