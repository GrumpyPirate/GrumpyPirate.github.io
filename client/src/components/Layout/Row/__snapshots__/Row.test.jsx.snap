// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> should render as expected, without crashing 1`] = `
<Row
  align={null}
  justify={null}
>
  <div>
    Test children
  </div>
</Row>
`;
